### DISCUSSION TITLE
Proper way to define constants in Platform IO?

### DISCUSSION THREAD

**Discussion by miesch1** [earlephilhower/ESP8266Audio#578] [Q&A]
Awesome library @earlephilhower, can't believe how useful and comprehensive it is!

So I spent longer than I would like to admit trying to get the newest version 1.9.7 to compile. I had no issues with 1.9.3. When my upgrade failed, I traced it back to the following line in AudioOutputULP.cpp:

`#ifdef ESP32`

that was replaced with this (among other places):

`#if CONFIG_IDF_TARGET_ESP32 || CONFIG_IDF_TARGET_ESP32S2 || CONFIG_IDF_TARGET_ESP32S3`

I didn't have to do anything special (besides define the board) to get `ESP32` defined. My question is, is there a better way to define `CONFIG_IDF_TARGET_ESP32` in Platform IO than in the platformio.ini itself? Is it possible to define this with an esp32 board as well? 

```
[env:esp32dev]
platform = espressif32
board = esp32dev
framework = arduino
lib_deps = earlephilhower/ESP8266Audio@^1.9.7

build_flags =
    -D CONFIG_IDF_TARGET_ESP32
```
This works fine, just wondering if that is the preferred way. Thanks for any thoughts,

-Mike

**Comment by nickolay** [comment]
FWIW https://github.com/plerup/espsoftwareserial/issues/219#issuecomment-876659600 suggests the appropriate CONFIG_IDF_TARGET_<..> defines are supposed to be set automatically via the sdkconfig.h. Indeed in my project `#if CONFIG_IDF_TARGET_ESP32S3` works without defining it manually (though I have not verified the minimal set of platformio.ini options that do it).
